jpa/src/main/java/org/example/app/PersonResource.java

```diff
@@ -18,159 +18,147 @@
         }
-        catch (Exception e){
-                e.printStackTrace();
-       }
 
         return people;
     }
 
     @GET
     @Path("/{personId}")
     public Person getPerson(@PathParam("personId") long id) {
-	Person person = (Person)em.find(Person.class, id);
+        Person person = (Person) em.find(Person.class, id);
         return person;
     }
 
     @POST
-    public String createPersonXHTML(@QueryParam("name") @NotEmpty @Size(min = 2, max = 50) String name,
-                             @QueryParam("age") @PositiveOrZero int age){
-            try{
-		Person p = new Person(name, age);
-                userTran.begin();
-                em.persist(p);
-                userTran.commit();
-                return "PersonList.xhtml?faces-redirect=true";
-            }
-            catch (Exception e){
-                e.printStackTrace();
-                return "CreatePerson.xhtml?faces-redirect=true";
-            }
+    public Long createPerson(@QueryParam("name") @NotEmpty @Size(min = 2, max = 50) String name,
+            @QueryParam("age") @PositiveOrZero int age) {
+        try {
+            Person p = new Person(name, age);
+            userTran.begin();
+            em.persist(p);
+            userTran.commit();
+            return p.id;
+        } catch (Exception e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
 
+    @PUT
+    @Path("/{personId}")
+    public String updatePerson(@PathParam("personId") long id, @Valid Person p) {
+        try {
+            userTran.begin();
+            Person person = em.find(Person.class, p.getId());
+            person.setName(p.name);
+            person.setAge(p.age);
+            em.merge(person);
+            userTran.commit();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        return "/PersonList.xhtml?faces-redirect=true";
     }
 
-    @POST
-    public Long createPerson(@QueryParam("name") @NotEmpty @Size(min = 2, max = 50) String name,
-                             @QueryParam("age") @PositiveOrZero int age){
-            try{
-                Person p = new Person(name, age);
-                userTran.begin();
-                em.persist(p);
-                userTran.commit();
-                return p.id;
-            }
-            catch (Exception e){
-                e.printStackTrace();
-                 return null;
-            }
+    @DELETE
+    @Path("/{personId}")
+    public String removePerson(@PathParam("personId") long id) {
+        Person deletePerson = em.find(Person.class, id);
+        try {
+            userTran.begin();
+            em.remove(em.merge(deletePerson));
+            userTran.commit();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        return "/PersonList.xhtml?faces-redirect=true";
+    }
 
+    public Collection<Person> getPeople() {
+        return people;
     }
 
+    public String createPersonXHTML(String name,int age) {
```

Comment:
Good that you removed `@POST`

Reply:
This was not just about removing the @POST. If you remove just the post, then all you have done is to swap one warning for another. The method needed to just be a simple method.

Reply:
Not sure what you mean... you mean you needed to remove the Bean Validation?